starlette/datastructures.py

```diff
@@ -438,7 +439,7 @@ async def write(self, data: bytes) -> None:
         if self.size is not None:
             self.size += len(data)
 
-        if self._in_memory:
+        if self._in_memory and self.file.tell() + len(data) <= getattr(self.file, "_max_size", sys.maxsize):
```

Comment:
what's this about?

Reply:
If the data to write makes the file exceed the `SpooledTemporaryFile` `_max_size`, `self.file.write` will do a blocking `rollover` operation.

Reply:
Other than accessing the private attribute this makes sense to me. Insofar as it's an optimization that fails gracefully I think it would be okay to do this.